staging/src/k8s.io/kubectl/pkg/cmd/debug/profile/applier.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type Applier interface {
```

Comment:
It looks like this design progressed from profiles that generate a "debugging object" to one where profiles mutate objects passed to them. I think that makes sense, and it made me realize that all of the `kubectl debug` invocations follow a pattern:

1. Generate a pod to create or update
2. Mutate based on CLI options
3. Mutate based on profile
4. Apply the pod changes

If the addition here is only to mutate via profile, then we can probably use a simpler interface, something like:

```
type applierFunc func(pod *corev1.Pod, containerName string, target runtime.Object) error

type Options struct {
	applier applierFunc
}

// New returns a new Options for the given profile name.
func New(profile string) (*Options, error) {
	switch profile {
	case "legacy":
		return &Options{Legacy}, nil
	}

	return nil, fmt.Errorf("unknown profile: %s", profile)
}

func (o *Options) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
	return o.applier(pod, containerName, target)
}
```

I pushed a full example here: https://github.com/verb/kubernetes/commit/dab06a57016203e645b07086553fe6f3c8af9901

what do you think?

Reply:
@verb Hi! I think it is good and have updated the code.